mmengine/runner/runner.py

```diff
@@ -1358,6 +1386,13 @@ def build_dataloader(dataloader: Union[DataLoader, Dict],
             # if `dataset_cfg` is not a valid type
             dataset = dataset_cfg
 
+        num_batch_per_epoch = dataloader_cfg.pop('num_batch_per_epoch', None)
+        if num_batch_per_epoch is not None:
+            world_size = get_world_size()
+            num_samples = num_batch_per_epoch * dataloader_cfg.get(
+                'batch_size', 1) * world_size
```

Comment:
Please refer to https://github.com/open-mmlab/mmengine/blob/170758aefe6cb05f61bf9353d03af1b8c1f4f73a/mmengine/runner/_flexible_runner.py#L1651 to obtain the bacth size.

Reply:
Got it. Thanks.